marklogic-data-hub/src/main/java/com/marklogic/hub/impl/HubConfigImpl.java

```diff
@@ -2096,4 +2102,32 @@ public void resetHubConfigs() {
         isHostLoadBalancer = null;
     }
 
+    public Object clone() throws CloneNotSupportedException{
+        HubConfigImpl hubConfig = (HubConfigImpl) super.clone();
+        ManageConfig manageConfig = new ManageConfig();
+        AppConfig appConfig = new AppConfig();
+        AdminConfig adminConfig = new AdminConfig();
+        hubConfig.setAdminConfig(adminConfig);
+        CmaConfig cmaConfig = null;
+        if (this.appConfig != null) {
+            cmaConfig = this.appConfig.getCmaConfig();
+        }
+        if (cmaConfig == null) {
+            cmaConfig = new CmaConfig(true);
+        }
+        appConfig.setCmaConfig(cmaConfig);
+        hubConfig.setAppConfig(appConfig, true);
+        hubConfig.setManageConfig(manageConfig);
```

Comment:
I don't think this will work, because none of the manageConfig properties are being set.

Reply:
The necessary values are set as part of the session creation in the one-ui project.

Reply:
What if when one-ui needs to create a new HubConfigImpl, it first creates one based on properties in the project, and then it overlays the user-specific stuff? 

I basically really want to avoid having implementations of "clone" in marklogic-data-hub that are based on one-ui requirements. I think "clone" is really only appropriate for objects with primitive fields and complex fields that are either stateless or have no user-specific state, and neither DataHubImpl nor HubConfigImpl fit those descriptions (particularly since we have circular dependencies between them, which is something else I'd like to resolve - HubConfigImpl should not have any knowledge of those 4 Spring-managed beans).